src/libstd/keyword_docs.rs

```diff
@@ -925,8 +925,42 @@ mod mut_keyword {}
 //
 /// Make an item visible to others.
 ///
-/// The documentation for this keyword is [not yet complete]. Pull requests welcome!
+///`pub` makes any module, function, or data structure accessible from inside
+/// of external modules.
+/// ```rust
+/// mod other_mod {
+///     pub mod public_module {
+///         pub struct PublicStruct;
+/// }
+///     let emptyStruct = other_mod::public_module::PublicStruct;
```

Comment:
Let's use a `pub use` statement here, to demonstrate that syntax and also avoid the compile error from a let in item context.

We may also need to add a `# fn main() {}` to make rustdoc happy.